cmd/codegen/generator/go/templates/src/_dagger.gen.go/defs.go.tmpl

```diff
@@ -244,10 +244,18 @@ func Connect(ctx context.Context, opts ...dagger.ClientOpt) (*Client, error) {
 		return nil, err
 	}
 
-	return &Client{
+	c := &Client{
 		query:  dag.QueryBuilder(),
 		client: dag.GraphQLClient(),
-	}, nil
+	}
+
+	{{ range $i, $dep := DependenciesRef }}
```

Comment:
Bleh, it's really confusing we have these near identical `Client`s. 

We should try and unify them - in `client.go.tmpl`. The `IsDevMode` statement should be in `client.go.tmpl`, so all the client code can be in one place.

Also, maybe a comment explaining why which one is which?